addons/website_slides/static/src/js/slides_course_quiz_finish.js

```diff
@@ -0,0 +1,152 @@
+odoo.define('website_slides.quiz.finish', function (require) {
+'use strict';
+
+var Dialog = require('web.Dialog');
+var core = require('web.core');
+var _t = core._t;
+
+var SlideQuizFinishModal = Dialog.extend({
+    template: 'slide.slide.quiz.finish',
+    events: {
+        "click .o_wslides_quiz_modal_btn": '_onClickNext',
+    },
+
+    init: function(parent, options) {
+        this.quiz = options.quiz;
+        this.hasNext = options.hasNext;
+        this.userId = options.userId;
+        options = _.defaults(options || {}, {
+            size: 'medium',
+            dialogClass: 'd-flex p-0',
+            $parentNode: $('.o_wslides_js_lesson_quiz'),
+            technical: false,
+            renderHeader: false,
+            renderFooter: false
+        });
+        this._super.apply(this, arguments);
+    },
+
+    willStart: function() {
+        var self = this;
+        return this._super.apply(this, arguments).then(function () {
+            self.$modal.addClass('o_wslides_quiz_modal pt-5');
+            self.$modal.find('.modal-dialog').addClass('mt-5');
+            self.$modal.find('.modal-content').addClass('shadow-lg');
+        });
+    },
+
+    start: function() {
+        var self = this;
+        this._super.apply(this, arguments).then(function () {
+            self.opened().then(function () {
+                self._animateProgressBar();
+                self._animateText()
+            });
+        });
+    },
+
+    //--------------------------------
+    // Handlers
+    //--------------------------------
+
+    _onClickNext: function() {
+        this.trigger_up('slide_go_next', {
+            widget: this
+        });
```

Comment:
I think the widget should handle its own destruction.
That way you can avoid passing "widget" like that, and also clean callers.

So first trigger_up, then call this.destroy().

I'm guessing that should work? Give it a try.